app/models/concerns/spree/gateway/adyen_gateway.rb

```diff
@@ -64,9 +66,14 @@ def handle_response(response, original_reference = nil)
       )
     end
 
+    def store_for_payment(psp_reference)
```

Comment:
It's pretty difficult to understand how this is intended to work, even after a couple passes. 

After about 10 minutes I now understand this is just finding the store that had a payment with this psp ref.

It would be great to see more explicit documentation as to the intent of this implementation, and why it's required.

In my opinion this is complex enough, and has enough independent concerns from this module that it should be moved into it's own class/module.